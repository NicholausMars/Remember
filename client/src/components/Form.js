import React, {Component} from 'react';
import { connect } from 'react-redux';
import { addCard } from '../actions';

export class Form extends Component{
  constructor(){
    super();
    this.state = {
          title:'TITLE...',
          description:''
    };
  }

  Add(){
    const {title, description} = this.state;
    if( title !== '' && description !== '')
      this.props.addCard(this.state);

    this.setState({
        title: '',
        description: ''
     });
  }

  render(){
    return (
           <div>
                <h1 className="heading-primary">Remember</h1>
                <div className="input__box">
                  <input
                    autoFocus
                    className="input__field"
                    type="text"
                    onChange={(event)=>{ this.setState({ title: event.target.value}) }}
                    value={this.state.title}
                  />

                  <textarea
                    className="input__textarea"
                    name="basic" rows="5" cols="50" maxLength="200"
                    onChange={(event)=>{ this.setState({ description: event.target.value }) }}
                    value={this.state.description}
                    >
                  </textarea>
                  <button onClick={this.Add.bind(this)}  className="btn btn__add">+</button>
                </div>
          </div>
        );
  }
}
export default connect(null, {addCard})(Form);
