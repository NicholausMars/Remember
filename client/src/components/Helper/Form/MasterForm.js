import React, {Component } from 'react';
import { connect } from 'react-redux';
import {updateMaster,showMasterForm} from '../../../actions';
class MasterForm extends Component{
  // pass a changable value to initialize state
  // state = { title : "" , visible:false};
  state = {visible:false};
  constructor(props){
    super(props);
    this.updateMasterCard =this.updateMasterCard.bind(this);
    this.inputRef = React.createRef();
  }

  componentWillReceiveProps(nextProps){
    if(nextProps.test.visible !== this.state.visible){
      // console.log(nextProps);
      this.setState((state, props)=>{
        return {visible: nextProps.test.visible};
      }) ;
    }
  }
  // componentDidUpdate(prevProps, prevState){
  // }
  updateMasterCard(event){
    event.preventDefault();
    let date = new Date().toDateString();
    this.props.updateMaster(this.props.test.id, { title:this.inputRef.current.value, date});
    this.props.showMasterForm(false);
  }

  render(){
    return (
      <div className="create__contrainer "  style={ this.state.visible ? { display: 'block'} : {display:'none'} }>
        <form className="create__cards  u-mt-sm">

          <div className="create__field">
              <input type="text" ref= {this.inputRef} placeholder="update" className="input__master-title" maxLength="32" />
          </div>
          <div className="create__cta u-mb-sm">
            <button onClick= {this.updateMasterCard} className="create__cta-btn "> Update Master Card </button>
          </div>
        </form>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    test: state.test
  };
}

 export default connect(mapStateToProps, {updateMaster,showMasterForm})(MasterForm);
