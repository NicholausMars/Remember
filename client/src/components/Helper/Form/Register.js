  import React, {PureComponent } from 'react'
  // import { Link } from 'react-router-dom'
  import {validatePsw} from  '../Validate';
  class Register extends PureComponent{

    constructor(props){
      super(props);
      this.state = {userName : "", pwd : "", pwdCheck: ""};
    }
  UpdateUserName = (e) =>{this.setState({userName: e.target.value}); }
  Updatepwd = (e) =>{this.setState({pwd: e.target.value});}
  UpdatepwdCheck = (e) =>{this.setState({pwdCheck: e.target.value});}


  onSubmit = (e) =>{
    e.preventDefault();
    const {userName ,pwd, pwdCheck} =  this.state;
    // console.log(this.state);
    if(userName !== "" && pwd !== "" && pwdCheck !== ""){
      // check if user name is already Their in server

      if(validatePsw(pwd,pwdCheck)){
        // send to server
        // console.log( pwd ,userName)
        console.log(" validation ");
      }else{
        console.log(" invalid password ");
      }
    }else{
      //
      console.log("A field has not been inputed");
    }

  }
  // onSubmit={this.onSubmit}
    render(){
      return(
        <div className="modal__form">
            <form className="form">
            <div className="u-mb-sm ">
              <h2 className="heading__quaternary">Register</h2>
            </div>
              <div className="form__group">
                <input id="name" className="form__input" type="text"  value={this.state.userName}  onChange={this.UpdateUserName}  maxLength="32" />
                <label className="form__label" htmlFor="name">UserName</label>
              </div>
              <div className="form__group">
                <input id="psw"  className="form__input" type="password" value={this.state.pwd} onChange={this.Updatepwd} />
                <label className="form__label" htmlFor="psw">Password</label>
              </div>
              <div className="form__group">
                <input id="checkpsw" className="form__input" type="password" value={this.state.pwdCheck} onChange={this.UpdatepwdCheck} />
                <label className="form__label" htmlFor="checkpsw">Password Check</label>
              </div>

              <div className="form__group--btn">
                <button className="form__btn form__btn--submit" onClick = {this.onSubmit}>Submit</button>
                <button className="form__btn form__btn--close" onClick= {this.props.onClose} >Close</button>
              </div>
            </form>
        </div>
      );
    }

  }

              // <Link className="" to="/">Submit</Link>
  export default Register;

  // this.userRef      = React.createRef();
  // this.passRef      = React.createRef();
  // this.passCheckRef = React.createRef();
