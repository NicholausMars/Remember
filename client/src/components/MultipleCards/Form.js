import React, {Component} from 'react';
import { connect } from 'react-redux';
import { addCard, updateForm, updateCard } from '../../actions';
import  uuidv1 from 'uuid/v1'
import PropTypes from 'prop-types';
/*
    I want the update method to use the same form.
    I want to make it obvious that the forms beening use for the update
*/

export class Form extends Component{
  constructor(props){
    super(props);
    this.state = {
      title:'',
      description:'',
      firstRender: true
    };
  }
  /*
    Place this.props.updateForm({}); to fix an issue with updating something else after adding
    problem with deleting a field as well
  */
componentWillReceiveProps(nextProps){
  // console.log("nextProps",nextProps);
  let {display, title, description} = nextProps.formupdate;
  if(display && this.state.firstRender){
    console.log(display,title,description );
    this.setState({firstRender: false, title, description});
  }
}
  Add = () => {
    if(validate(this.state))
      this.props.addCard({id:uuidv1() ,...this.state});
      this.emptyState();
      //fixes a problem after you ad the card of an updated field
      this.props.updateForm({});
  }
  Update = () =>{
    let {id} = this.props.formupdate;
    let {title, description} = this.state;
    this.props.updateCard({id,title,description});
    this.emptyState();
    this.props.updateForm({});
  }
  emptyState = () => {
    this.setState({  title: '',description: '',firstRender: true});
  }

  render(){
    let {display} = this.props.formupdate;
    return (
           <div>
                <div className="input__box">
                  <input
                    autoFocus
                    className="input__field"
                    type="text"
                    onChange={(event)=>{ this.setState({ title: event.target.value}) }}
                    value={this.state.title}
                    maxLength="32"
                  />
                  <textarea
                    className="input__textarea"
                    name="basic" rows="5" cols="50" maxLength="200"
                    onChange={(event)=>{ this.setState({ description: event.target.value }) }}
                    value={this.state.description}
                    >
                  </textarea>
                  <button onClick={this.Update}  className={display?"btn btn__update btn__update-show":"btn btn__update btn__update-hide"}>U</button>
                  <button onClick={this.Add}  className="btn btn__add">+</button>
                </div>
          </div>
        );
  }
}
Form.propTypes ={
  addCard: PropTypes.func
}

export const mapStateToProps = (state) => {
  return {
    formupdate: state.formupdate
  };
}
export const validate = ({title,description }) => {
    if( title !== '' && description !== '') return true;
    return false;
}
export default connect(mapStateToProps, {addCard, updateForm, updateCard})(Form);

//https://medium.com/backticks-tildes/testing-your-react-component-with-jest-and-enzyme-276eef45bea0

// for lifecycle methods
//https://engineering.musefind.com/react-lifecycle-methods-how-and-when-to-use-them-2111a1b692b1
