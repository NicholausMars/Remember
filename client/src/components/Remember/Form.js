import React, {Component} from 'react';
import { connect } from 'react-redux';
import { addCard } from '../../actions';
import  uuidv1 from 'uuid/v1'
import PropTypes from 'prop-types';
/*
  I want the update method to use the same form.
   I want to make it obvious that the forms beening use for the update
*/

export class Form extends Component{
  constructor(props){
    super(props);
    this.state = {
      title:'',
      description:''
    };
  }

  Add = () => {

    if(validate(this.state))
      this.props.addCard({id:uuidv1() ,...this.state});
      this.emptyState();
  }
  emptyState = () => {
    this.setState({  title: '',description: ''});
  }

  render(){
    // const {display, id, title, description} = this.props.formupdate;
    // console.log(this.props.formupdate);
    return (
           <div>
                <div className="input__box">
                  <input
                    autoFocus
                    className="input__field"
                    type="text"
                    onChange={(event)=>{ this.setState({ title: event.target.value}) }}
                    value={this.state.title}
                  />
                  <textarea
                    className="input__textarea"
                    name="basic" rows="5" cols="50" maxLength="200"
                    onChange={(event)=>{ this.setState({ description: event.target.value }) }}
                    value={this.state.description}
                    >
                  </textarea>
                  {
                    /*
                    steps:
                    1.render button
                    2. when button is click
                    3. update forUpdate display: false card to be null
                    4. use the update card method to remove and add a new card
                    5. end of update
                    */
                  }
                  <button onClick={this.Add}  className="btn btn__add">+</button>
                </div>
          </div>
        );
  }
}
Form.propTypes ={
  addCard: PropTypes.func
}

export const mapStateToProps = (state) => {
  return {
    formupdate: state.formupdate
  };
}
export const validate = ({title,description }) => {
    if( title !== '' && description !== '') return true;
    return false;
}
export default connect(mapStateToProps, {addCard})(Form);

//https://medium.com/backticks-tildes/testing-your-react-component-with-jest-and-enzyme-276eef45bea0
