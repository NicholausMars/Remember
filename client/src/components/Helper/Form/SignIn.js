import React, {Component} from 'react'
import { Link } from 'react-router-dom'
// import google from  '../../../img/SVG/google-plus3.svg'
// import facebook from  '../../../img/SVG/facebook2.svg'
import Modal from '../Modal'
import {Register} from '../Form'
/*
  import {Modal} from '../Modal';
  <div className="modal">
    <Modal>
      <h1>Register</h1>
      <input type="text" placeholder="userName" value=""/>
      <input type="password" placeholder="password" value=""/>
      <input type="password" placeholder="password" value=""/>
    </Modal>
  </div>
*/

export class SignIn extends Component{
  constructor(props){
    super(props);
    this.state = {
      userName: "Username",
      password: "password",
      isOpen: false
    };
  }
  toggleModal = (e) => {
    e.preventDefault();
    this.setState({
      isOpen: !this.state.isOpen
    });
  }

  render(){
    // console.log("Modal : ",this.state.isModal);
    return (

      <div className="formA">
        <form>
          <div className=" u-mb-sm u-mt-md">
            <h2 className="heading__secondary">
              Sign In
            </h2>
          </div>

          <div className="formA__group">
            <input
             onChange={(event)=>{this.setState({userName: event.target.value })}}
              type="text" className="formA__input"
               value={this.state.userName} /* allows the input to depend on the state*/
               required/>

          </div>

          <div className="formA__group">
            <input
             onChange={(event)=>{this.setState({password: event.target.value })}}
              type="password"
              className="formA__input"
               value={this.state.password}  required/>
          </div>
          <div className="formA__group">

            <Link className="btnA btnA--green " to="/main">Sign in &rarr;</Link>
            {/*<a className="btnA btnA--green " href="/auth/google"  to="/auth/login">Sign in &rarr;</a>*/}
          </div>
        </form>

            <button  className="btn__register" onClick={this.toggleModal}>Register</button>
            <a href="/auth/google" className="btnA__svg">
              <svg  className="btnA__svg--google"   version="1.1" xmlns="http://www.w3.org/2000/svg"  viewBox="0 0 30 30">
                <path d="M16 0c-8.838 0-16 7.162-16 16s7.162 16 16 16 16-7.163 16-16-7.163-16-16-16zM12 24c-4.425 0-8-3.575-8-8s3.575-8 8-8c2.162 0 3.969 0.787 5.363 2.094l-2.175 2.088c-0.594-0.569-1.631-1.231-3.188-1.231-2.731 0-4.963 2.263-4.963 5.050s2.231 5.050 4.963 5.050c3.169 0 4.356-2.275 4.538-3.45h-4.537v-2.744h7.556c0.069 0.4 0.125 0.8 0.125 1.325 0 4.575-3.063 7.819-7.681 7.819zM26 16v2h-2v-2h-2v-2h2v-2h2v2h2v2h-2z"></path>
              </svg>
            </a>
            <a href="/auth/facebook" className="btnA__svg">
              <svg  className="btnA__svg--facebook"  version="1.1" xmlns="http://www.w3.org/2000/svg"  viewBox="0 0 30 30">
                <path d="M29 0h-26c-1.65 0-3 1.35-3 3v26c0 1.65 1.35 3 3 3h13v-14h-4v-4h4v-2c0-3.306 2.694-6 6-6h4v4h-4c-1.1 0-2 0.9-2 2v2h6l-1 4h-5v14h9c1.65 0 3-1.35 3-3v-26c0-1.65-1.35-3-3-3z"></path>
              </svg>
            </a>
            <Modal show={this.state.isOpen}>
                      <Register onClose={this.toggleModal}/>
            </Modal>

      </div>
    );
  }
// <img src={google} alt='google'/>
  // <img src={facebook} alt='facebook'/>

    // onClose={this.toggleModal}
}
// onClick={this.ToggleModel}/
// {this.state.isModal && <Modal > <Register  /> </Modal>}
