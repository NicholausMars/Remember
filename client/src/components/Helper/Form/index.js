import Form from './Form'
import MasterForm from './MasterForm'
import Register from './Register'
import {SearchBar} from './SearchBar'
import  {SignIn} from './SignIn'

export {
  Form,
  MasterForm,
  Register,
  SearchBar,
  SignIn
}
