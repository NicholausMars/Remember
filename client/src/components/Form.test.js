import React from 'react';
import {  shallow, configure } from 'enzyme';
import {Form}  from './Form';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

describe('Form', ()=>{
  const form = shallow(<Form/>);
  test('Form render correctly', () => expect(form).toMatchSnapshot());
  test('redner input field', ()=>{
    console.log(form.debug());
  })
});
