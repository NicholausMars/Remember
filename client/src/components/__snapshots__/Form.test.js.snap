// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Form render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="heading-primary"
>
          Remember
</h1>,
        <div
          className="input__box"
>
          <input
                    autoFocus={true}
                    className="input__field"
                    onChange={[Function]}
                    type="text"
                    value="TITLE..."
          />
          <textarea
                    className="input__textarea"
                    cols="50"
                    maxLength="200"
                    name="basic"
                    onChange={[Function]}
                    rows="5"
                    value=""
          />
          <button
                    className="btn btn__add"
                    onClick={[Function]}
          >
                    +
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Remember",
          "className": "heading-primary",
        },
        "ref": null,
        "rendered": "Remember",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              autoFocus={true}
              className="input__field"
              onChange={[Function]}
              type="text"
              value="TITLE..."
/>,
            <textarea
              className="input__textarea"
              cols="50"
              maxLength="200"
              name="basic"
              onChange={[Function]}
              rows="5"
              value=""
/>,
            <button
              className="btn btn__add"
              onClick={[Function]}
>
              +
</button>,
          ],
          "className": "input__box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoFocus": true,
              "className": "input__field",
              "onChange": [Function],
              "type": "text",
              "value": "TITLE...",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input__textarea",
              "cols": "50",
              "maxLength": "200",
              "name": "basic",
              "onChange": [Function],
              "rows": "5",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "+",
              "className": "btn btn__add",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "+",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="heading-primary"
>
            Remember
</h1>,
          <div
            className="input__box"
>
            <input
                        autoFocus={true}
                        className="input__field"
                        onChange={[Function]}
                        type="text"
                        value="TITLE..."
            />
            <textarea
                        className="input__textarea"
                        cols="50"
                        maxLength="200"
                        name="basic"
                        onChange={[Function]}
                        rows="5"
                        value=""
            />
            <button
                        className="btn btn__add"
                        onClick={[Function]}
            >
                        +
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Remember",
            "className": "heading-primary",
          },
          "ref": null,
          "rendered": "Remember",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                autoFocus={true}
                className="input__field"
                onChange={[Function]}
                type="text"
                value="TITLE..."
/>,
              <textarea
                className="input__textarea"
                cols="50"
                maxLength="200"
                name="basic"
                onChange={[Function]}
                rows="5"
                value=""
/>,
              <button
                className="btn btn__add"
                onClick={[Function]}
>
                +
</button>,
            ],
            "className": "input__box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoFocus": true,
                "className": "input__field",
                "onChange": [Function],
                "type": "text",
                "value": "TITLE...",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input__textarea",
                "cols": "50",
                "maxLength": "200",
                "name": "basic",
                "onChange": [Function],
                "rows": "5",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+",
                "className": "btn btn__add",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "+",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
